### PR TITLE
govukapp-912:  Add configurations for alpha/beta builds

### PR DESCRIPTION
**Pull request by joshdubey**
Added AppEnvironment service to provide config info. 
Created Debug, Alpha, and Release configs
Created corresponding GovUKDev, GovUKAlpha, and GovUK run schemes
Added run phase for copying correct Firebase configuration into place

The GovUKDev scheme is to be used to build and run locally, and will sign with a development cert
The GovUKAlpha scheme signs the app with an Ad Hoc certificate for Firebase deployment
The GovUK scheme signs the app with a distribution certificate for Testflight deployment

Repurposed alpha github action to deploy to Firebase.  Created new beta action to deploy to testflight (essentially the old alpha action, except it will trigger when a release branch is created)
Made fastlane modifications to enable distribution to firebase

### CODE REVIEW

File: Fastlane/.build.yml [hunk 1]
```diff
@@ -26,7 +28,22 @@ alpha:
   <<: *DEFAULTS
   app:
     <<: *APP_DEFAULTS
+    scheme: GovUKAlpha
```
Comment (batesyuk) [inline Fastlane/.build.yml:31]: Is it worth changing the default scheme to `GovUKDev`?

File: Production/govuk_ios/SupportingFiles/Info.plist [hunk 2]
```diff
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>Base URL</key>
```
Comment (batesyuk) [inline Production/govuk_ios/SupportingFiles/Info.plist:5]: Can we have the key without spaces, please? Either snake or pascal case. Probably Pascal as the other keys in Info.plist are Pascal.

File: Production/govuk_ios/Resources/Config/Release.xcconfig [hunk 3]
```diff
@@ -0,0 +1 @@
+BASE_URL = https:/$()/app.integration.publishing.service.gov.uk
```
Comment (batesyuk) [inline Production/govuk_ios/Resources/Config/Release.xcconfig:1]: Live url?

File: Fastlane/.build.yml (hunk 1 above)
Comment (joshdubey) [reply Fastlane/.build.yml:31]: Tested with running fastlane unit locally and things working okay as is, so as discussed we can leave for now

**Review by batesyuk — APPROVED** [review]

**Pull request merged**